src/hotspot/share/utilities/vmError.cpp

```diff
@@ -1443,10 +1443,9 @@ void VMError::report_and_die(int id, const char* message, const char* detail_fmt
     install_secondary_signal_handler();
   } else {
 #if defined(_WINDOWS)
-    // If UseOSErrorReporting we call this for each level of the call stack
-    // while searching for the exception handler.  Only the first level needs
-    // to be reported.
-    if (UseOSErrorReporting && log_done) return;
+    if (UseOSErrorReporting && log_done) {
+      raise_fail_fast(_siginfo, _context);
+    }
```

Comment:
I think this bit might not be needed anymore.  I think the reason for the old code here is to deal with
the repeated calls to report_and_die as we walk up the Windows exception filter list and return because
of UseOSErrorReporting.  We don't do that anymore.  I think we could instead now not do anything here
and treat repeated calls "normally", reporting them as such and eventually dying, either via Windows
fail-fast or abort.

Reply:
I think we should remove this part. Otherwise concurrent threads running into faults cause the VM to end abruptly while the primary thread is its in the last stages of error reporting: after log_done is set but before os::die was called (we do some things here: JFR shutdown, NMT statistics, compiler replay file writing).

In fact, this may be an existing bug which could explain missing/corrupt replay files or JFR files if UseOSErrorReporting is set.

Reply:
Yeah, I was approaching the same conclusion.  I'll remove it.